From 858a61be6eb7758b8b8901e12a81de322efddefc Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 26 Apr 2021 18:33:23 +0300
Subject: [PATCH] [SW-37130] pldm: add hl_spi_baudr_get() definition

Added function hl_spi_baudr_get() from greco_arc
to avoid compilation issue. spi_init() which inturn calls
hl_spi_baudr_get() is now called from common function.

Change-Id: I60b9c8040b6a61fa9c301e2388cb696c3b927420
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/greco_arc_gnu/pldm.c b/greco_arc_gnu/pldm.c
index 99f94ef..d84dfc2 100644
--- a/greco_arc_gnu/pldm.c
+++ b/greco_arc_gnu/pldm.c
@@ -66,4 +66,12 @@
 	preboot_uart_init(UART_CLOCK_FREQUENCY / PLDM_UART_BAUDRATE);
 }
 
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor = 2; /* Min Divider */
+
+	/* baud rate doesn't support odd number */
+	return (factor + 1) & 0xFFFE;
+}
+
 void preboot_init_pcie_phy(void) {}
